Fix typos in semgrep-langs readme (#17)
returntocorp/semgrep-langs: README.md

JSON schema is:

<dl>
  <dt>id</dt><dd>Unique string identifier.</dd>
  <dt>name</dt><dd>Human-readable name.</dd>
  <dt>keys</dt><dd>All possible values that can reference a language when used as an
    argument to `--lang`; should be non-empty; keys must map one-to-one
    to languages.</dd>
  <dt>exts</dt><dd>File extensions that Semgrep scans for this language; should be
    non-empty.</dd>
  <dt>excluded_exts</dt><dd>File extensions that Semgrep skips even if one of exts is a
    suffix.</dd>
  <dt>example_ext</dt><dd>File extension of the example language. Specify if

>     length of exts is greater than one. Must start with a period.</dd>

Reviewer: My understanding is that this field is meant to hold the most idiomatic extension for the files in the language. This extension is the one that will be used by the Semgrep app where a file name needs to be shown to the user (examples, downloads, and whatnot). For C++, we might choose `.cc` while both `.cc` and `.cpp` are commonly used by C++ programmers.

Author: Interesting! Creating a new suggestion based on this information. @mjambon

>   <dt>example_ext</dt><dd>File extension of the example language. Specify if
>     length of exts is greater than one. Must start with a period.</dd>

Author: Should I also specify that this extension will be used by Semgrep App?

```suggestion
  <dt>example_ext</dt><dd>Most common extension of the example language.
    Specify example_ext if the language has more than one exts.
    Must start with a period.</dd>
```

Author: @mjambon I like that you used the word idiomatic, which is now included in the suggestion above. :-)

  <dt>reverse_exts</dt><dd>If present, overrides exts when determining _what_ language a
    file is, otherwise defaults to exts; reverse_exts must be a one-to-one
    mapping to languages.</dd>
  <dt>maturity</dt><dd>One of "ga", "beta", "alpha", or "develop"; "alpha" displays as "Experimental"

>     in documentation; "develop" is not displayed in documentation.</dd>

Author: I am not sure if that was the meaning of: "is hidden altogether"

Reviewer: I'm eyeballing this and it looks like the very last `</dd>` is mismatched. I don't know why we're using HTML markup in the first place. I don't think it's justified. Let's switch to Markdown, which won't have this problem.

Reviewer: Regarding the meaning of "hidden altogether": I'm not sure either. The context is missing. "Undocumented" seems about right to me.

Reviewer: For the formatting, I'd make the following translation:

```
<dt>id</dt><dd>Unique string identifier.</dd>
```
->
```
* **id**: Unique string identifier.
```
If you don't do it this time, please remind me to do it later.

Author: Markdown would not allow us to use description lists, unfortunately. I would stick with HTML here, description list is really justified in our case, see: https://developers.google.com/style/lists ("_A set of terms, each with a description, definition, or explanation. Use this type of list if you want to draw attention to two or more terms (such as a glossary). The following is an example of a description list:_")

Reviewer: Please read about [KISS](https://en.wikipedia.org/wiki/KISS_principle). We can't afford to argue about this.

Author: @mjambon I did not find HTML description lists so complicated, but this document is here for devs, so I will change them to what you suggest above. The only note I have here is that the description list looks really nicer for this use case, but the source code is more verbose.
